packages/node/src/parseForPolicy.js

```diff
@@ -19,88 +18,81 @@ const {
   parse,
   inspectImports,
   codeSampleFromAstNode,
+  inspectEsmImports,
 } = require('lavamoat-tofu')
 const { checkForResolutionOverride } = require('./resolutions')
 
+/**
+ * @callback ResolveHook
+ * @param {string} requestedName
+ * @param {string} referrer
+ * @returns {string|null}
+ */
+
 // file extension omitted can be omitted, eg https://npmfs.com/package/yargs/17.0.1/yargs
-const commonjsExtensions = ['', '.js', '.cjs']
-const resolutionOmittedExtensions = ['.js', '.json']
+const jsExtensions = new Set(
+  /** @type {const} */ (['.cjs', '.mjs', '.js', '.json'])
+)
+const cjsResolutionOmittedExtensions = /** @type {const} */ (['.js', '.json'])
 
 /**
  * Allow use of `node:` prefixed builtins.
  */
-const builtinPackages = [
-  ...builtinModules,
-  ...builtinModules.map((id) => `node:${id}`),
-]
-
-// approximate polyfill for node builtin
-const createRequire = (url) => {
-  return {
-    resolve: (requestedName) => {
-      const basedir = path.dirname(url.pathname)
-      const result = resolve.sync(requestedName, {
-        basedir,
-        extensions: resolutionOmittedExtensions,
-      })
-      // check for missing builtin modules (e.g. 'worker_threads' in node v10)
-      // the "resolve" package resolves as "worker_threads" even if missing
-      const resultMatchesRequest = requestedName === result
-      if (resultMatchesRequest) {
-        const isBuiltinModule = builtinPackages.includes(result)
-        const looksLikeAPath = requestedName.includes('/')
-        if (!looksLikeAPath && !isBuiltinModule) {
-          const errMsg = `Cannot find module '${requestedName}' from '${basedir}'`
-          const err = new Error(errMsg)
-          err.code = 'MODULE_NOT_FOUND'
-          throw err
-        }
-      }
-      // return resolved path
-      return result
-    },
-  }
+const BUILTIN_MODULES = Object.freeze(
+  new Set(
+    /** @type {const} */ ([
+      ...builtinModules,
+      ...builtinModules.map((id) => `node:${id}`),
+    ])
+  )
+)
+
+function isBuiltin(name) {
+  return BUILTIN_MODULES.has(name)
```

Comment:
we could use `require('node:module').isBuiltin(name)`, which works with both `node:`-prefixed paths and without.  I added this myself!!!